chain/client/src/sync.rs

```diff
@@ -629,7 +637,11 @@ impl StateSync {
                 shard_id,
                 false,
             ) {
-                Some(AccountOrPeerIdOrHash::AccountId(validator_stake.account_id.clone()))
+                if me.as_ref().map(|me| me != &validator_stake.account_id).unwrap_or(false) {
```

Comment:
Why `unwrap_or(false)`? If you are not a validator you would not request state from a validator?

Reply:
It makes sense.
Should it be `if me.as_ref().map(|me| me != &validator_stake.account_id).unwrap_or(true)`?

Reply:
Yes.